spec/java_buildpack/container/tomcat/tomcat_geode_store_spec.rb

```diff
@@ -0,0 +1,125 @@
+# Cloud Foundry Java Buildpack
+# Copyright 2013-2017 the original author or authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'spec_helper'
+require 'component_helper'
+require 'java_buildpack/container/tomcat/tomcat_geode_store'
+
+describe JavaBuildpack::Container::TomcatGeodeStore do
+  include_context 'component_helper'
+
+  let(:component_id) { 'tomcat' }
+
+  let(:configuration) do
+    { 'database'             => 'test-database',
+      'timeout'              => 'test-timeout',
+      'connection_pool_size' => 'test-connection-pool-size' }
+  end
+
+  it 'does not detect without a session-replication service' do
+    expect(component.detect).to be_nil
+  end
+
+  it 'does nothing when it does not detect',
+     app_fixture:   'container_tomcat_geode_store',
+     cache_fixture: 'stub-geode-store.tar' do
+
+    component.compile
+
+    expect(sandbox + "lib/stub-geode-store/stub-jar-1.jar").not_to exist
+    expect(sandbox + "lib/stub-geode-store/stub-jar-2.jar").not_to exist
+  end
+
+  context 'when there is a session-replication service' do
+    before do
+      allow(services).to receive(:one_service?).with(/session-replication/, 'locators', 'users')
+        .and_return(true)
```

Comment:
You'll want to run `bundle exec rake` to ensure that your code meets all formatting requirements.  The Rubocop task will give you pointers on where the code needs to change.